### ISSUE TITLE
Private key file should not be readable for anyone

### ISSUE THREAD

**Issue by Inndy** [Neilpang/acme.sh#40]
```
www-data@my-host:~/.le/my.host.com$ ls -lh
total 20K
-rw-r--r-- 1 www-data www-data 1.7K Jan 27 17:30 ca.cer
-rw-r--r-- 1 www-data www-data 2.1K Jan 27 17:30 my.host.com.cer
-rw-r--r-- 1 www-data www-data  487 Jan 27 17:30 my.host.com.conf
-rw-r--r-- 1 www-data www-data 1.6K Jan 27 17:30 my.host.com.csr
-rw-r--r-- 1 www-data www-data 3.2K Jan 27 17:30 my.host.com.key
    ^^^^^^ WTFFFFFF
```

Generated private key can be read by any user on this machine which is not secure.

**Comment by lucasRolff** [comment]
Since the files are stored within the root directory.
No it cannot be read by any user.
The directory `.le` has 0550 permissions - so even if you as a user on the system tries to read files below this directory, you'll get a permission denied..

See:
```
Perms:
wheel@server [~/.le]# sudo stat /root/.le/XXXX.dk/XXXX.dk.key
  File: `/root/.le/XXXX.dk/XXXX.dk.key'
  Size: 1675      	Blocks: 8          IO Block: 4096   regular file
Device: 10304h/66308d	Inode: 8687060     Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root

----

backupbo@XXXX.dk [~]# cat /root/.le/XXXX.dk/XXXX.dk.key
cat: /root/.le/XXXX.dk/XXXX.dk.key: Permission denied
```

Now we make it even worse:

```
Perms:
wheel@server [~/.le]# sudo stat /root/.le/XXXX.dk/XXXX.dk.key
  File: `/root/.le/XXXX.dk/XXXX.dk.key'
  Size: 1675      	Blocks: 8          IO Block: 4096   regular file
Device: 10304h/66308d	Inode: 8687060     Links: 1
Access: (0777/-rwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2016-01-17 12:12:08.875905886 +0100
Modify: 2016-01-17 12:12:08.870905886 +0100
Change: 2016-01-27 10:46:20.221942582 +0100

----

backupbo@XXXX.dk [~]# cat /root/.le/XXXX.dk/XXXX.dk.key
cat: /root/.le/XXXX.dk/XXXX.dk.key: Permission denied
```

**Comment by naodesu** [comment]
acme.sh can be used not only by root.
On some distributions/setups home directory of ordinary user can have 711 or even 755 access rights.
So, my suggestion is to use 700 access rights for ~/.acme directory, just like ~/.ssh and others.

**Comment by Neilpang** [comment]
@Inndy  @lucasRolff  @velum @naodesu 
fixed  bfdf1f482e0b012f0be88f271d1fe1fe37947ec1

we have 
```
umask 077
```
